src/main/java/nl/tudelft/lifetiles/graph/controller/TileController.java

```diff
@@ -65,15 +65,15 @@ public final Group drawGraph() {
      * set a certain changed property with the arguments.
      *
      * @param propertyName
-     *            - Name of the property
+     *            Name of the property
      * @param args
-     *            - Arguments for the function that needs to be called.
+     *            Arguments for the function that needs to be called.
      */
     public final void setChangedProperty(final String propertyName,
             final List<Object> args) {
 
         if (propertyName.equals("COLOR_CHANGED")) {
-            viewVar.changeVertexColour((Vertex) args.get(0),
+            viewVar.changeVertexColour((VertexView) args.get(0),
```

Comment:
The casting is prone to throw throw an error if the class' post-condition (first argument a `VertexView`, second one a `Color`). You should check this sooner re-throw it as a `IllegalArgumentException` with an appropriate message.